packages/shared/lib/services/connections/credentials/refresh.integration.test.ts

```diff
@@ -0,0 +1,292 @@
+import { beforeAll, describe, expect, it, vi } from 'vitest';
+
+import { multipleMigrations } from '@nangohq/database';
+import { logContextGetter, migrateLogsMapping } from '@nangohq/logs';
+import { Err, Ok, wait } from '@nangohq/utils';
+
+import { refreshOrTestCredentials } from './refresh.js';
+import { createConfigSeed, createConnectionSeed, seedAccountEnvAndUser } from '../../../seeders/index.js';
+import encryptionManager from '../../../utils/encryption.manager';
+import { NangoError } from '../../../utils/error.js';
+import connectionService from '../../connection.service.js';
+
+describe('refreshOrTestCredentials', () => {
+    beforeAll(async () => {
+        await multipleMigrations();
+        await migrateLogsMapping();
+    });
+
+    it('should not refresh if unauthenticated', async () => {
+        const { env, account } = await seedAccountEnvAndUser();
+        const integration = await createConfigSeed(env, 'unauthenticated', 'unauthenticated');
+        const connection = await createConnectionSeed({ env, provider: 'unauthenticated' });
+        const decryptedConnection = encryptionManager.decryptConnection(connection);
+        if (!decryptedConnection) {
+            throw new Error('Failed to decrypt connection');
+        }
+
+        await wait(2);
```

Comment:
why waiting?

Reply:
because otherwise it executes too fast, and some dates can have the same seconds. I'll add a comment